modules/edge-ren/src/lib.rs

```diff
@@ -0,0 +1,464 @@
+
+			Self::deposit_event(RawEvent::RenTokenDeleted(_ren_token_id));
+			Ok(())
+		}
+
+
+		/// Transfers tokens from the pallet account to a specified account
+		#[weight = 10_000]
+		fn spend_tokens(
+			origin,
+			#[compact] _ren_token_id: T::RenVMTokenIdType,
+			who: T::AccountId,
+			amount: BalanceOf<T>,
+		) -> DispatchResult
+		{
+			T::ControllerOrigin::ensure_origin(origin)?;
+
+			let asset_id = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?.ren_token_asset_id;
+
+			T::Assets::transfer(asset_id, Self::account_id().into(), who.clone(), amount)?;
+
+			Self::deposit_event(RawEvent::RenTokenSpent(_ren_token_id, amount));
+			Ok(())
+		}
+
+		/// Mints the token, after fee deduction, if the signature provided is valid over the provided arguments
+		#[weight = 10_000]
+		fn mint(
+			origin,
+			#[compact] _ren_token_id: T::RenVMTokenIdType,
+			who: T::AccountId,
+			p_hash: [u8; 32],
+			amount: BalanceOf<T>,
+			n_hash: [u8; 32],
+			sig: EcdsaSignature,
+		) -> DispatchResult
+		{
+			ensure_none(origin)?;
+
+			let ren_token = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?;
+
+			ensure!(ren_token.ren_token_mint_enabled, Error::<T>::RenTokenMintDisabled);
+
+			let asset_id = ren_token.ren_token_asset_id;
+
+			let mint_fee_value = Permill::from_parts(ren_token.ren_token_mint_fee).mul_floor(amount);
+
+			// Skipped checking if the mint call for fee might overflow or cause min_balance error
+			// as that should not prevent a user from his operation
+
+			// MINT CALL for user
+			T::Assets::mint(asset_id, who.clone(), (amount - mint_fee_value).into())?;
+			// Attempt mint call for fees
+			T::Assets::mint(asset_id, Self::account_id().into(), mint_fee_value.into());
+
+			Signatures::insert(&sig, ());
+			Self::deposit_event(RawEvent::RenTokenMinted(_ren_token_id, who, amount));
+			Ok(())
+		}
+
+
+		/// Burns the token so that the corresponding pegged currency can be released to the "to" address by RenVM
+		/// Checks if the token is bitcoin by name, "renBTC" (or "renTestBTC" for testnet) and validates the "to" address accordingly
+		/// The actual amount burnt is calculated based on the burn fee, and is the value of the corresponding pegged currency that the user will be granted from RenVM
+		/// The burn fee is transfered from the users account to the pallets account
+		#[weight = 10_000]
+		fn burn(
+			origin,
+			#[compact] _ren_token_id: T::RenVMTokenIdType,
+			to: DestAddress,
+			amount: BalanceOf<T>,
+		) -> DispatchResult
+		{
+			let sender = ensure_signed(origin)?;
+
+			let ren_token = RenTokenRegistry::<T>::get(&_ren_token_id).ok_or_else(|| Error::<T>::RenTokenNotFound)?;
+
+			ensure!(ren_token.ren_token_burn_enabled, Error::<T>::RenTokenBurnDisabled);
+
+		 	let asset_id = ren_token.ren_token_asset_id;
+
+
+			match sp_std::str::from_utf8(ren_token.ren_token_name.as_slice()).map_err(|_| Error::<T>::UnexpectedError) {
```

Comment:
This doesn't look general enough to where it supports assets that are created using this pallet.

Reply:
Could you please elaborate on your concern.  I do not see what breaks generality here.

The line 317 is accessing the stored id used by the assets pallet, which I will merge with ren_token_id (https://github.com/hicommonwealth/edgeware-node/pull/272#discussion_r548383646)
The line 320 is a part of the validation mechanism for the "to" address. The match statement helps select the correct validator on the basis of "ren_token.ren_token_name",  after converting it into a string literal.